### PR TITLE
[AMORO-3139] Improve list content of optimizing page

### PR DESCRIPTION
**Pull request by chouchouji**
## Why are the changes needed?

Link issue #3139.

## Brief change log

-

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (yes / no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: amoro-web/mock/modules/optimize.js [hunk 1]
```diff
@@ -37,6 +37,15 @@ export default [
       result: ['container1', 'container2']
     }),
   },
+  {
+    url: '/mock/ams/v1/optimize/action',
+    method: 'get',
+    response: () => ({
+      code: 200,
+      msg: 'success',
+      result: ['major', 'minor', 'clean']
```
Comment (majin1102) [inline amoro-web/mock/modules/optimize.js:46]: Here shall add pending, idle

File: amoro-web/mock/modules/optimize.js [hunk 2]
```diff
@@ -62,13 +71,14 @@ export default [
               }
             },
             "tableName": "test_catalog.db.user",
-            "optimizeStatus": "idle",
+            "status": "full_optimizing.commtting",
```
Comment (majin1102) [inline amoro-web/mock/modules/optimize.js:74]: Let's postpone this feature for now and keep this PR as clean and simple as possible.

File: amoro-web/mock/modules/optimize.js [hunk 3]
```diff
@@ -62,13 +71,14 @@ export default [
...
             "duration": 1195501081,
             "fileCount": 0,
             "fileSize": 0,
             "quota": 0.1,
             "quotaOccupation": 0.0,
-            "groupName": "local"
+            "groupName": "local",
+            "resource": "8c 16g"
```
Comment (majin1102) [inline amoro-web/mock/modules/optimize.js:81]: Let's postpone this feature for now and keep this PR as clean as possible.

File: amoro-web/src/services/optimize.service.ts [hunk 4]
```diff
@@ -22,17 +22,23 @@ export function getOptimizerGroups() {
...
+
 export function getOptimizerTableList(
   params: {
     optimizerGroup: string
     dbSearchInput: string
     tableSearchInput: string
     page: number
     pageSize: number
+    action: string[]
+    sortField: string
```
Comment (majin1102) [inline amoro-web/src/services/optimize.service.ts:37]: I think we shall need sortField as well as sortOrder for ordering feature.

Anyway Let's postpone this sort feature for now(preserve action) and keep this PR as clean as possible.

File: amoro-web/src/types/common.type.ts [hunk 5]
```diff
@@ -263,18 +263,18 @@ export interface ITableIdentifier {
 }
 export interface IOptimizeTableItem {
   tableName: string
-  optimizeStatus: string
+  status: string
```
Comment (majin1102) [inline amoro-web/src/types/common.type.ts:266]: This modification could be done in next PR

File: amoro-web/src/utils/table.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+  * distributed under the License is distributed on an "AS IS" BASIS,
+  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+  */
+
+/**
+ * Calculate table max width
+ */
+export function getTableMaxWidth(columns: Array<{ width: number }>) {
```
Comment (majin1102) [inline amoro-web/src/utils/table.ts:22]: What's this for?

File: amoro-web/src/views/optimize/components/List.vue [hunk 7]
```diff
@@ -44,24 +45,38 @@ const loading = ref<boolean>(false)
...
-  { dataIndex: 'quotaOccupationDesc', title: t('occupation'), width: 120, ellipsis: true },
+const columns = computed(() => [
+  { dataIndex: 'tableName', title: t('table'), width: 200, scopedSlots: { customRender: 'tableName' } },
+  { dataIndex: 'groupName', title: t('optimizerGroup'), width: 180, ellipsis: true },
+  { dataIndex: 'status', title: t('status'), width: 240, ellipsis: true },
+  { dataIndex: 'duration', title: t('duration'), width: 150, ellipsis: true, sorter: true },
+  { dataIndex: 'fileCount', title: t('fileCount'), width: 150, ellipsis: true },
+  { dataIndex: 'fileSizeDesc', title: t('fileSize'), width: 150, ellipsis: true },
+  { dataIndex: 'quota', title: t('quota'), width: 150, ellipsis: true },
+  { dataIndex: 'resource', title: t('resource'), width: 150, ellipsis: true, sorter: true },
```
Comment (majin1102) [inline amoro-web/src/views/optimize/components/List.vue:56]: resource could be added in next PR

**Review by majin1102 — CHANGES_REQUESTED** [review]
Hi, chouchouji

Thanks for your contribution.
I left some comments, and I have spiltted the original issue into more subtasks so that front and backend could better follow up.
I suggest you make this PR clean to this subtask: https://github.com/apache/amoro/issues/3234

**Comment by majin1102** [comment]
> Hi, chouchouji
> 
> Thanks for your contribution. I left some comments, and I have spiltted the original issue into more subtasks so that front and backend could better follow up. I suggest you make this PR clean to this subtask: #3234

File: amoro-web/src/views/optimize/components/List.vue [hunk 8]
```diff
@@ -84,6 +99,8 @@ async function getTableList() {
       tableSearchInput: tableSearchInput.value || '',
       page: pagination.current,
       pageSize: pagination.pageSize,
+      action: action.value,
```
Comment (majin1102) [inline amoro-web/src/views/optimize/components/List.vue:102]: Is this meant to be actions?

**Review by majin1102 — CHANGES_REQUESTED** [review]

**Comment by majin1102** [comment]
The backend work is: https://github.com/apache/amoro/pull/3236

File: amoro-web/src/utils/table.ts (hunk 6 above)
Comment (chouchouji) [reply amoro-web/src/utils/table.ts:22]: This is a function that scrolls table if it is too long

https://github.com/user-attachments/assets/e64b7868-f08d-4bba-a628-f825760db921

File: amoro-web/src/views/optimize/components/List.vue (hunk 7 above)
Comment (chouchouji) [reply amoro-web/src/views/optimize/components/List.vue:56]: ok, I will remove resource field.

File: amoro-web/mock/modules/optimize.js (hunk 2 above)
Comment (chouchouji) [reply amoro-web/mock/modules/optimize.js:74]: I get it.

File: amoro-web/src/views/optimize/components/List.vue (hunk 8 above)
Comment (chouchouji) [reply amoro-web/src/views/optimize/components/List.vue:102]: Yes, it is the data selected by action multiple select.

File: amoro-web/src/services/optimize.service.ts (hunk 4 above)
Comment (chouchouji) [reply amoro-web/src/services/optimize.service.ts:37]: Do you mean I should keep action and remove sortField?

**Comment by chouchouji** [comment]
@majin1102  Thanks for your review! I had updated code. Please check it when you are free.

File: amoro-web/src/services/optimize.service.ts (hunk 4 above)
Comment (majin1102) [reply amoro-web/src/services/optimize.service.ts:37]: yes

**Review by majin1102 — CHANGES_REQUESTED** [review]
Thanks for feeding back
I left one comment, PTAL

File: amoro-web/mock/modules/optimize.js [hunk 9]
```diff
@@ -37,6 +37,15 @@ export default [
       result: ['container1', 'container2']
     }),
   },
+  {
+    url: '/mock/ams/v1/optimize/action',
```
[the remark this answers is not in the document]
Comment (chouchouji) [reply amoro-web/mock/modules/optimize.js:41]: ok, i get it.

File: amoro-web/src/services/optimize.service.ts (hunk 4 above)
Comment (chouchouji) [reply amoro-web/src/services/optimize.service.ts:37]: If we don't need sortOrder now, I will delete it.

**Review by majin1102 — APPROVED** [review]
LGTM

**Pull request merged**